### DISCUSSION TITLE
Using Isar and Freezed

### DISCUSSION THREAD

**Discussion by jersonal-com** [isar/isar#864] [Help]
I am trying to create a freezed model that works with Isar as well. 

When running a quite minimalistic code like this:

```@freezed
@Collection()
class Test with _$Test {
  const factory Test({
    int? id,
    String? name,
  }) = _Test;
}
```
I get the following error message:

```
Unsupported type. Please annotate the property with @ignore.
package:test/test.freezed.dart:23:27
   ╷
23 │   $TestCopyWith<Test> get copyWith => throw _privateConstructorUsedError;
   │                           ^^^^^^^^
   ╵
```
Would appreciate any help in what I am doing wrong. Freezed is V 2.1.0 and Isar is on 3.0.2.

**Comment by DevNico** [comment]
You can use `@Collection(ignore: {'copyWith'})` even though I would recommend having separate models for your database layer and mapping between them and your (freezed) domain layer.
